src/main/java/org/osc/manager/rest/server/api/ManagerConnectorApis.java

```diff
@@ -0,0 +1,209 @@
+@Consumes({ MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML })
+@Produces({ MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML })
+
+public class ManagerConnectorApis {
+
+    private static final Logger logger = Logger.getLogger(ManagerConnectorApis.class);
+    private EntityManager em;
+    private TransactionControl txControl;
+
+    /**
+     * Creates the DB handler
+     *
+     *
+     */
+    public void createManagerConnectorDb(EntityManager em, TransactionControl txControl) throws Exception {
+        this.em = em;
+        this.txControl = txControl;
+    }
+
+    /**
+     * Creates the MC
+     *
+     * @return MC
+     * @throws Exception
+     */
+    @POST
+    public ApplianceManagerConnectorEntity createManagerConnector(ApplianceManagerConnectorEntity entity)
+            throws Exception {
+
+        logger.info("Creating Manager Connector...");
+
+        return this.txControl.required(new Callable<ApplianceManagerConnectorEntity>() {
+
+            @Override
+            public ApplianceManagerConnectorEntity call() throws Exception {
+                ApplianceManagerConnectorEntity mc = new ApplianceManagerConnectorEntity();
+                mc.setName(entity.getName());
+                mc.setid(entity.getid());
+                ManagerConnectorApis.this.em.persist(mc);
+                return mc;
+            }
+        });
+    }
+
+    /**
+     * Updates MC for a given ID
+     *
+     * @return - updated MC
+     */
+    @Path("/{applianceManagerConnectorId}")
+    @PUT
+    public ApplianceManagerConnectorEntity updateManagerConnector(@PathParam("applianceManagerConnectorId") Long amcId,
+            ApplianceManagerConnectorEntity entity) {
+
+        logger.info("Updating MC Entity ...");
+
+        return this.txControl.required(new Callable<ApplianceManagerConnectorEntity>() {
+            @Override
+            public ApplianceManagerConnectorEntity call() throws Exception {
+                CriteriaBuilder cb = ManagerConnectorApis.this.em.getCriteriaBuilder();
+
+                CriteriaQuery<ApplianceManagerConnectorEntity> q = cb.createQuery(ApplianceManagerConnectorEntity.class);
+                Root<ApplianceManagerConnectorEntity> r = q.from(ApplianceManagerConnectorEntity.class);
+                q.select(r).where(cb.and(cb.equal(r.get("id"), amcId)));
+
+                ApplianceManagerConnectorEntity result = ManagerConnectorApis.this.em.createQuery(q).getSingleResult();
+                result.setName(entity.getName());
+                ManagerConnectorApis.this.em.persist(result);
+                return entity;
+            }
+        });
+    }
+
+    /**
+     * Delete MC for a given ID
+     *
+     * @return - deleted MC
+     */
+    @Path("/{applianceManagerConnectorId}")
+    @DELETE
+    public ApplianceManagerConnectorEntity deleteManagerConnector(@PathParam("applianceManagerConnectorId") Long amcId,
```

Comment:
no need to return the entity deleted. Should just return status ok and empty body